File: library/src/main/java/renetik/android/CSApplication.java
package renetik.android;

import android.app.Application;

import androidx.annotation.NonNull;

import static renetik.android.lang.CSLang.info;
import static renetik.android.lang.CSLang.model;

public class CSApplication extends Application {

    private static CSApplication application;

    public CSApplication() {
        application = this;
    }

    @NonNull
    public static CSApplication application() {
        return application;
    }

    public void onLowMemory() {
        super.onLowMemory();
        info("onLowMemory");
        model().logger().onLowMemory();
    }

    public void onTerminate() {
        super.onTerminate();
        info("onTerminate");
    }
}

File: library/src/main/java/renetik/android/CSApplication.kt
package renetik.android

import android.app.Application
import renetik.android.lang.CSDoLater
import renetik.android.lang.CSLang.info
import renetik.android.lang.CSLogger
import renetik.android.model.CSValueStore

lateinit var application: CSApplication

abstract class CSApplication : Application() {

    abstract val name: String
    abstract val logger: CSLogger
    open val store: CSValueStore by lazy { CSValueStore("ApplicationSettings") }
    abstract val isDebugBuild: Boolean

    override fun onCreate() {
        super.onCreate()
        CSDoLater.initializeHandler()
        application = this
    }

    override fun onLowMemory() {
        super.onLowMemory()
        info("onLowMemory")
        logger.onLowMemory()
    }

    override fun onTerminate() {
        super.onTerminate()
        info("onTerminate")
    }

}
